static/app/views/releases/detail/commitsAndFiles/preprodBuilds.tsx

```diff
@@ -0,0 +1,303 @@
+                                #{build.vcs_info.head_sha?.slice(0, 6) || 'N/A'}
+                              </Text>
+                              <Text size="sm" variant="muted">
+                                -
+                              </Text>
+                              <Text size="sm" variant="muted">
+                                {build.vcs_info.head_ref || 'main'}
+                              </Text>
+                            </Flex>
+                          </Flex>
+                        </SimpleTable.RowCell>
+
+                        <SimpleTable.RowCell>
+                          {build.size_info
+                            ? formatBytesBase10(build.size_info.install_size_bytes)
+                            : '-'}
+                        </SimpleTable.RowCell>
+
+                        <SimpleTable.RowCell>
+                          {build.size_info
+                            ? formatBytesBase10(build.size_info.download_size_bytes)
+                            : '-'}
+                        </SimpleTable.RowCell>
+
+                        <SimpleTable.RowCell>
+                          {build.app_info.date_added ? (
+                            <TimeSince
+                              date={build.app_info.date_added}
+                              unitStyle="short"
+                            />
+                          ) : (
+                            '-'
+                          )}
+                        </SimpleTable.RowCell>
+                      </Link>
+                    </SimpleTable.Row>
+                  ))}
+                </SimpleTableWithColumns>
+              </PanelBody>
+            </Panel>
+            {pagination && (
+              <Flex direction="row" gap="md" align="center" justify="end" paddingTop="md">
+                <Text variant="muted" size="md">
+                  {t(
+                    'Page %s of %s',
+                    pagination.page + 1,
+                    Math.ceil(
+                      (typeof pagination.total_count === 'number'
+                        ? pagination.total_count
+                        : 0) / pagination.per_page
+                    )
+                  )}
+                </Text>
+                <ButtonBar merged gap="0">
+                  <Button
+                    icon={<IconChevron direction="left" />}
+                    aria-label={t('Previous')}
+                    size="sm"
+                    disabled={!pagination.has_prev}
+                    onClick={() => setPage((pagination.prev || 0) + 1)}
+                  />
+                  <Button
+                    icon={<IconChevron direction="right" />}
+                    aria-label={t('Next')}
+                    size="sm"
+                    disabled={!pagination.has_next}
+                    onClick={() => {
+                      setPage((pagination.next || pagination.page + 1) + 1);
+                    }}
+                  />
```

Comment:
There's a Pagination component that can help a lot with this: https://github.com/getsentry/sentry/blob/master/static/app/components/pagination.tsx#L74